internal/pkg/service/common/configmap/visit.go

```diff
@@ -12,18 +12,18 @@ type OnField func(field reflect.StructField, path orderedmap.Path) (fieldName st
 
 type OnValue func(vc *VisitContext) error
 
-type AfterValue func(vc *VisitContext) error
-
 type VisitConfig struct {
 	// InitNilPtr initializes each found nil pointer with an empty struct.
+	// The operation is performed before all nested field are processed.
 	InitNilPtr bool
+	// EmptyStructToNilPtr replaces each found empty struct with nil pointer, if possible.
+	// The operation is performed after all nested field are processed.
+	EmptyStructToNilPtr bool
 	// OnField maps field to a custom field name, for example from a tag.
 	// If ok == false, then the field is ignored.
 	OnField OnField
 	// OnValue is called on each field
 	OnValue OnValue
-	// AfterValue is called after all nested fields are processed, if any. Can be nil.
-	AfterValue AfterValue
 }
```

Comment:
I replaced `AfterValue` setting with the more specific `EmptyStructToNilPtr` setting.